src/block.rs

```diff
@@ -0,0 +1,314 @@
+            paysplit: 0.5,
+            vote: 0,
+            treasury: 286_810_000_000_000_000,
+            coinbase: 0,
+        };
+    }
+}
+
+impl Block {
+    /// Receives the a publickey and the previous block hash
+    ///
+    /// * `block_creator` - `secp256k1::PublicKey` of the block creator
+    /// * `prevbsh` - Previous block hash in bytes
+    pub fn new(creator: PublicKey, prevbsh: [u8; 32]) -> Block {
+        return Block {
+            body: BlockBody::new(creator, prevbsh),
+            bsh: [0; 32],
+        };
+    }
+
+    /// Creates a block solely from the block body. Used when
+    /// deserializing a block either from disk or from the network
+    ///
+    /// * `body` - `BlockBody` of new `Block`
+    pub fn from_block_body(body: BlockBody) -> Block {
+        return Block {
+            body: body,
+            bsh: [0; 32],
+        };
+    }
+
+    /// Returns the `Block` id
+    pub fn id(&self) -> u64 {
+        self.body.id
+    }
+
+    /// Returns the `Block` timestamp
+    pub fn timestamp(&self) -> u64 {
+        self.body.ts
+    }
+
+    /// Returns the previous `Block` hash
+    pub fn previous_block_hash(&self) -> [u8; 32] {
+        self.body.previous_block_hash
+    }
+
+    /// Returns the `Block` creator's `secp256k1::PublicKey`
+    pub fn creator(&self) -> PublicKey {
+        self.body.creator
+    }
+
+    /// Returns the `Block`'s `Transaction`s
+    pub fn txs(&self) -> &Vec<Transaction> {
+        &self.body.txs
+    }
+
+    /// Returns the `Block` burnfee
+    pub fn burnfee(&self) -> u64 {
+        self.body.burnfee
+    }
+
+    /// Returns the `Block` difficulty
+    pub fn difficulty(&self) -> f32 {
+        self.body.difficulty
+    }
+
+    /// Returns the `Block` vote
+    pub fn vote(&self) -> i8 {
+        self.body.vote
+    }
+
+    /// Returns the `Block` paysplit
+    pub fn paysplit(&self) -> f32 {
+        self.body.paysplit
+    }
+
+    /// Returns the `Block` treasury
+    pub fn treasury(&self) -> u64 {
+        self.body.treasury
+    }
+
+    /// Returns the `Block` coinbase
+    pub fn coinbase(&self) -> u64 {
+        self.body.coinbase
+    }
+
+    /// Generate the block hash
+    ///
+    /// TODO -- extend list of information we use to calculate the block hash
+    pub fn block_hash(&self) -> [u8; 32] {
+        let mut data: Vec<u8> = vec![];
+
+        let id_bytes: [u8; 8] = unsafe { transmute(self.body.id.to_be()) };
+        let ts_bytes: [u8; 8] = unsafe { transmute(self.body.ts.to_be()) };
+        let cr_bytes: Vec<u8> = self.body.creator.serialize().iter().cloned().collect();
+
+        data.extend(&id_bytes);
+        data.extend(&ts_bytes);
+        data.extend(&cr_bytes);
+
+        return hash(data);
+    }
+
+    /// Converts our blockhash from a byte array into a hex string
+    pub fn block_hash_hex(&self) -> String {
+        hex::encode(&self.block_hash())
+    }
+
+    /// Swaps a list of transactions already in memory into blocks transactions
```

Comment:
"...into blocks transactions"? is that a typo?

Reply:
yeah I'll clean up the wording on some of the documentation